### DISCUSSION TITLE
Instead of calling the icon without a path, is it possible to set a custom path in the plugin code?

### DISCUSSION THREAD

**Discussion by MiltonBalaOfficial** [xournalpp/xournalpp#6114] [Q&A]
Currently, the folder of a plugin  has to be placed in `plugin` folder and icons have to be placed in `icon` folder. Is it possible to set a  custom path for plugin icons within the code? If it is possible then we don't have to place the icons separately, just placing the folder containing the code and icons would be enough to use the plugin.

**Comment by rolandlo** [comment]
No, it's not possible. Icons in Gtk are always referred to by name, not by path. They are searched for in a list of directories specified by the search path of the icon theme. See https://docs.gtk.org/gtk3/method.IconTheme.set_search_path.html

**Reply by MiltonBalaOfficial** [reply]
Thanks for the reply. I was eagerly waiting for a reply whether it is a "yes" or "no"

**Comment by 6xthFNGR** [comment]
@rolandlo The issue came up due to X++ in Snap:
First of all: Ubuntu PPA is stuck in version 1.2.2-2build3, which is quite old
Snap offers 1.2.4, so its preferable for now:

But due to snap sandbox, some critical folders are not accessible and mounted as read-only.

I do have write privileges to the "simulated" Plugin folder ( ~/snap/xournalpp/100/.config/xournalpp/plugins )
but X++ (snap) and the Plugins can not access the native ( ~/.local/share/icons/hicolor/scalable/ ) folder where the Icons belong,
neither do I as user/superuser have write privileges to anything beyond ~/snap/xournalpp/100/.local/share/icons. Dont ask me why the policy is to grant access to ~/snap/xournalpp/100/.config but not ~/snap/xournalpp/100/.local/share/icons/ and if this behaviour is caused by snap or X++ policies.

While X++ (snap) can access native home directory ( ~/ ), it cannot access hidden folders within that for security reasons (hidden folder = DOT Foldername, e.g. ".local")
One workaround on my side would be to create messy duplications of the icons folders or create additional empty folders and use symlink, but I rather want to keep everything contained and not clutter my ~/ (home) directory.

Maybe thats something X++ devs can help with?

**Comment by rolandlo** [comment]
What Ubuntu version are you using? Which PPA do you use (stable or unstable)? It shouldn't be stuck on 1.2.2 unless you have an old Ubuntu version.

I didn't know that there are no write privileges to `~/snap/xournalpp/current/.local/share/icons/hicolor/scalable`. I can investigate that.

**Reply by rolandlo** [reply]
It looks like one can simply delete the `~/snap/xournalpp/current/.local/share/icons/hicolor` folder (which just contains symlinks to a read-only folder) and create it new.

![Bildschirmfoto vom 2024-12-11 08-50-47](https://github.com/user-attachments/assets/a231d69d-a5a5-4ce8-bd80-6c8d618b76a3)

I don't know why those symlinks were installed in the first place, but they don't look necessary at all. The snap runs fine without them.

**Comment by 6xthFNGR** [comment]
@rolandlo 
Ubuntu 24 LTS - newest LTS available and just released this year.
PPA: Official Ubuntu repo, so "stable". Im not a huge fan of adding custom repos if it can be avoided.
